lib/_tls_common.js

```diff
@@ -153,6 +153,38 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const engineKey = options.engineKey;
+  const privateKeyEngine = options.privateKeyEngine;
+  if (engineKey && privateKeyEngine) {
+    if ((typeof options.engineKey === 'string') &&
+        (typeof options.privateKeyEngine === 'string')) {
+      if (c.context.setEngineKey)
+        c.context.setEngineKey(engineKey, privateKeyEngine);
+      else
+        throw new ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED();
+    } else if (typeof options.engineKey != 'string') {
+      throw new ERR_INVALID_ARG_TYPE('options.engineKey',
+                                     ['string', 'null', 'undefined'],
+                                     options.engineKey);
+    } else {
+      throw new ERR_INVALID_ARG_TYPE('options.privateKeyEngine',
+                                     ['string', 'null', 'undefined'],
+                                     options.privateKeyEngine);
+    }
+  }
+
+  if (options.sigalgs) {
+    if (typeof options.sigalgs !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('options.sigalgs', 'string',
+                                     options.sigalgs);
+    }
+
+    if (c.context.setClientSigalgs)
+      c.context.setClientSigalgs(options.sigalgs);
```

Comment:
As above, don't extract the property multiple times. Doing that allows to bypass the type validation.